toolkit/tools/internal/pkgjson/pkgjson.go

```diff
@@ -73,11 +74,49 @@ type Package struct {
 	RunTests      bool          `json:"RunTests"`      // Should we run tests for this package.
 }
 
+func packageLess(a, b PackageVer) bool {
+	if a.Name == b.Name {
+		v1 := versioncompare.New(a.Version)
+		v2 := versioncompare.New(b.Version)
+		return v1.Compare(v2) < 0
+	}
+
+	return a.Name < b.Name
+}
+
+// SortPackageList ensures the package list is sorted in a deterministic way. It will primarily sort by name, then
+// version. It will also recursively sort the package lists (requires, buildRequires, testRequires) of each package.
```

Comment:
I'd drop the word "recursively", since we don't go down each (build|test)requires and sort their (build|test)requires and then their (build|test)requires and then their (build|test)requires and... :)
```suggestion
// version. It will also sort the package lists (requires, buildRequires, testRequires) of each package.
```